### PR TITLE
 Merge the JsonValueType enum into JsonTokenType

### PR DESCRIPTION
**Pull request by ahsonkhan**

### CODE REVIEW

File: src/System.Text.JsonLab/System/Text/Json/JsonType.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.JsonLab
+{
+    // Choosing the relevant subset of JsonTokenType since we can only support 8 (2 ^ 3)
+    // Keep the enum order in sync with JsonTokenType since we case between the two
```
Comment (JamesNK) [inline src/System.Text.JsonLab/System/Text/Json/JsonType.cs]: case -> cast

**Comment by ahsonkhan** [comment]
@dotnet-bot test this please

File: src/System.Text.JsonLab/System/Text/Json/SpanExtensions.cs [hunk 2]
```diff
@@ -14,25 +14,20 @@ public static string ConvertToString(this ReadOnlySpan<byte> span)
             return Encodings.Utf8.ToString(span);
         }
 
-        private static bool IsDecimal(this ReadOnlySpan<byte> span)
-        {
-            return span.IndexOfAny((byte)'.', (byte)'e', (byte)'E') != -1;
-        }
-
-        public static (NumberType, object) GetNumberType(this ReadOnlySpan<byte> span)
```
Comment (JamesNK) [inline src/System.Text.JsonLab/System/Text/Json/SpanExtensions.cs]: Is NumberType used anywhere else?

Comment (ahsonkhan) [reply src/System.Text.JsonLab/System/Text/Json/SpanExtensions.cs]: Nope. it can be removed.

**Pull request merged**